test/topo/topo.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	utils "github.com/onosproject/onos-topo/test/utils"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+var log = logging.GetLogger("topo")
+
+// TestAddRemoveDevice adds devices to the storage, lists and checks that they are in database and removes devices from the storage
+func (s *TestSuite) TestAddRemoveDevice(t *testing.T) {
+
+	log.Debugf("Creating connection")
+	conn, err := utils.CreateConnection()
+	assert.NilError(t, err)
+	log.Debugf("Creating Topo Client")
+	client := topoapi.NewTopoClient(conn)
+
+	log.Debugf("Adding first device to the topo store")
+	_, err = client.Create(context.Background(), &topoapi.CreateRequest{
+		Object: &topoapi.Object{
+			ID:   "1",
+			Type: topoapi.Object_ENTITY,
+		},
+	})
+	assert.NilError(t, err)
+
+	log.Debugf("Adding second device to the topo store")
+	_, err = client.Create(context.Background(), &topoapi.CreateRequest{
+		Object: &topoapi.Object{
+			ID:   "2",
+			Type: topoapi.Object_ENTITY,
+		},
+	})
+	assert.NilError(t, err)
+
+	log.Debugf("Checking whether added device exists")
+	gres, err := client.Get(context.Background(), &topoapi.GetRequest{
+		ID: "1",
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, topoapi.ID("1"), gres.Object.ID)
+
+	log.Debugf("Listing all devices")
+	res, err := client.List(context.Background(), &topoapi.ListRequest{})
+	assert.NilError(t, err)
+	log.Debugf("Verifying that there are two devices stored")
+	assert.Equal(t, len(res.Objects) == 2 &&
+		(res.Objects[0].ID == "1" || res.Objects[1].ID == "1"), true)
+	//assert.Condition(t, func() bool {
+	//	return len(res.Objects) == 2 &&
+	//		(res.Objects[0].ID == "1" || res.Objects[1].ID == "1")
+	//})
+
+	log.Debugf("Updating first device")
+	obj := gres.Object
+	//obj.Attributes = make(map[string]string)
```

Comment:
You can use this utility functions associated with an object to Set/Get aspects and include them in tests: 
https://github.com/onosproject/onos-api/blob/master/go/onos/topo/topo.go